servers/zts/src/main/java/com/yahoo/athenz/zts/cert/impl/JDBCCertRecordStoreConnection.java

```diff
@@ -37,24 +34,35 @@
 
     private static final String SQL_GET_X509_RECORD = "SELECT * FROM certificates WHERE provider=? AND instanceId=? AND service=?;";
     private static final String SQL_INSERT_X509_RECORD = "INSERT INTO certificates " +
-            "(provider, instanceId, service, currentSerial, currentTime, currentIP, prevSerial, prevTime, prevIP, clientCert) " +
-            "VALUES (?, ?,?,?,?,?,?,?,?,?);";
+            "(provider, instanceId, service, currentSerial, currentTime, currentIP, prevSerial, prevTime, prevIP, clientCert, " +
+            "lastNotifiedTime, lastNotifiedServer, expiryTime, hostName) " +
+            "VALUES (?, ?,?,?,?,?,?,?,?,?,?,?,?,?);";
     private static final String SQL_UPDATE_X509_RECORD = "UPDATE certificates SET " +
-            "currentSerial=?, currentTime=?, currentIP=?, prevSerial=?, prevTime=?, prevIP=?" +
+            "currentSerial=?, currentTime=?, currentIP=?, prevSerial=?, prevTime=?, prevIP=?, " +
+            "lastNotifiedTime=?, lastNotifiedServer=?, expiryTime=?, hostName=?" +
             "WHERE provider=? AND instanceId=? AND service=?;";
     private static final String SQL_DELETE_X509_RECORD = "DELETE from certificates " +
             "WHERE provider=? AND instanceId=? AND service=?;";
     private static final String SQL_DELETE_EXPIRED_X509_RECORDS = "DELETE FROM certificates " +
             "WHERE currentTime < ADDDATE(NOW(), INTERVAL -? MINUTE);";
-    
-    public static final String DB_COLUMN_SERVICE        = "service";
-    public static final String DB_COLUMN_CURRENT_IP     = "currentIP";
-    public static final String DB_COLUMN_CURRENT_SERIAL = "currentSerial";
-    public static final String DB_COLUMN_CURRENT_TIME   = "currentTime";
-    public static final String DB_COLUMN_PREV_IP        = "prevIP";
-    public static final String DB_COLUMN_PREV_SERIAL    = "prevSerial";
-    public static final String DB_COLUMN_PREV_TIME      = "prevTime";
-    public static final String DB_COLUMN_CLIENT_CERT    = "clientCert";
+    private static final String SQL_UPDATE_UNREFRESHED_X509_RECORDS_NOTIFICATION_TIMESTAMP = "UPDATE certificates SET lastNotifiedTime=?, lastNotifiedServer=? " +
+            "WHERE currentTime < (CURRENT_DATE - INTERVAL 1 DAY) AND DATEDIFF(CURRENT_TIME, currentTime) IN (1,7,14,21,28) AND (lastNotifiedTime IS NULL || lastNotifiedTime < (CURRENT_DATE - INTERVAL 1 DAY))";
+    private static final String SQL_LIST_NOTIFY_UNREFRESHED_X509_RECORDS = "SELECT * FROM certificates WHERE lastNotifiedTime=? AND lastNotifiedServer=?;";
+
```

Comment:
Currently implemented logic is to send notifications for certificates that didn't refresh for more than a day. Notification will be sent once a week for a month.
We can change this logic in a future pull request to consider expiration for example.

Reply:
Note this logic was fixed in the up-coming pull request according to our last session. You may ignore it for now.